### ISSUE TITLE
AoE performance issues

### ISSUE THREAD

**Issue by ChrisNZL** [ChrisNZL/Tallowmere2#40]
Lightning, Flash of Light, Giant's Toe -- when the game checks for nearby creatures, performance suffers.

**Comment by ChrisNZL** [comment]
Some improvement: was calling `FixedUpdate` for some things that should have just been called in `Update`, resulting in multiple calls per frame with `FixedUpdate`.

**Comment by ChrisNZL** [comment]
Final investigation. Updated engine to Unity 2017.4.12f1 - no change in performance.

There were just too many blood puddles stacking on top of each other, which was affecting performance with gibs and coins entering and exiting from the puddle triggers. Even tried making the puddles not be triggers; no effect.

There _was_ a minor performance improvement if I used CircleCollider2D instead of PolygonCollider2D for the puddles, but not by much.

I've added a check to not have a puddle be instantiated if there are already 4 puddles overlapping the same spot. This seems to resolve the performance issues now for v0.0.4.
